### PR TITLE
Fix/detect objects fix

### PR DESCRIPTION
**Pull request by EricChen-Lei**
# PR Details
## Description

changed detected objects payload to match CDASim's format and added few new object_type
## Related Jira Key
CDAR-446

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sensorlib/src/objects/DetectedObject.py [hunk 1]
```diff
@@ -45,19 +45,24 @@ def build_detected_object(carla_actor, allowed_semantic_tags):
         if (bounding_box == None):
             return None
         
+        projection_string = projection_string_config
+
+        size_x = carla_actor.bounding_box.extent.x*2
```
Comment (paulbourelly999) [inline sensorlib/src/objects/DetectedObject.py:50]: Why are we multiplying by 2 here?

File: sensorlib/src/sensor/SemanticLidarSensor.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def compute_detected_objects(self):
         timestamp, hitpoints = self.__data_collector.get_carla_lidar_hitpoints()
 
         # Compute data needed for occlusion operation
-        actor_angular_extents = self.compute_actor_angular_extents(detected_objects)
+        #actor_angular_extents = self.compute_actor_angular_extents(detected_objects)
```
Comment (paulbourelly999) [inline sensorlib/src/sensor/SemanticLidarSensor.py:104]: Are we removing this because we removed orientation? Lets leave a TODO comment here stating that we commented this out because we are assuming velocity vector represents orientation

File: sensorlib/src/sensor/SemanticLidarSensor.py [hunk 3]
```diff
@@ -509,7 +509,7 @@ def update_object_metadata_from_hitpoint(self, obj, hitpoints, timestamp):
             new_position = np.subtract(obj.position, np.array([sensor_location.x, sensor_location.y, sensor_location.z]))
 
         return replace(obj,
-                       object_type=new_object_type,
-                       timestamp=timestamp,
+                       type=new_object_type,
+                       #timestamp=timestamp,
```
Comment (paulbourelly999) [inline sensorlib/src/sensor/SemanticLidarSensor.py:513]: Why is this no longer necessary

File: sensorlib/src/util/CarlaUtils.py [hunk 4]
```diff
@@ -88,11 +99,15 @@ def determine_object_type(carla_actor, allowed_semantic_tags):
         #using type_id instead of semantic_tags
         #issue with semantic_tags in version 0.9.10:https://github.com/carla-simulator/carla/issues/2161
         temp_id_list =  carla_actor.type_id.split(".")
-        temp_id = temp_id_list[0] 
+        temp_id = temp_id_list[0]
         if temp_id == "vehicle":
-            return "Vehicles"
+            vehicle_type = temp_id_list[2]
+            if vehicle_type in CarlaUtils.vehicle_lookup_table.keys():
```
Comment (paulbourelly999) [inline sensorlib/src/util/CarlaUtils.py:105]: Please add conditions for TRUCK, CYCLIST, VAN, and MOTORCYCLE

File: sensorlib/src/util/CarlaUtils.py [hunk 5]
```diff
@@ -105,14 +120,14 @@ def get_semantic_tag_name(tag_id):
...
 
-    @staticmethod
-    def get_semantic_tag_id(tag_name):
-        """
-        Get the integer semantic tag ID for a given tag name.
-        :param tag_name: The string tag name to look up.
-        :return: The tag ID.
-        """
-        return carla.CityObjectLabel.names.get(tag_name, 0)
+    # @staticmethod
```
Comment (paulbourelly999) [inline sensorlib/src/util/CarlaUtils.py:123]: Why are we commenting this out. Please leave a TODO comment with reasoning for this.

File: sensorlib/src/util/SimulatedSensorUtils.py [hunk 6]
```diff
@@ -18,18 +18,21 @@ def default(self, obj):
...
+                'projString': obj.projString,
+                'objectId': obj.objectId,
+                'position':{'x': obj.position[0], 'y': obj.position[1], 'z': obj.position[2]},
+                'positionCovariance': obj.positionCovariance.tolist(),
+                'velocity':{'x': obj.velocity[0], 'y': obj.velocity[1], 'z': obj.velocity[2]},
+                'velocityCovariance': obj.velocityCovariance.tolist(),
+                'angularVelocity': {'x': obj.angularVelocity[0], 'y': obj.angularVelocity[1], 'z': obj.angularVelocity[2]},
+                'angularVelocityCovariance': obj.angularVelocityCovariance.tolist(),
+                'size': {'length': obj.size[0], 'height': obj.size[1], 'width': obj.size[2]}
+                #'timestamp': obj.timestamp,
```
Comment (paulbourelly999) [inline sensorlib/src/util/SimulatedSensorUtils.py:32]: We need timestamp, why is this removed

File: sensorlib/src/objects/DetectedObject.py (hunk 1 above)
Comment (paulbourelly999) [reply sensorlib/src/objects/DetectedObject.py:56]: Can we remove the 2 here. I think we are not sure about this calculation so lets not do any arbitrary multiplication

File: sensorlib/config/simulated_sensor_config.yaml [hunk 7]
```diff
@@ -19,4 +23,5 @@ lidar_sensor:
   channels: 32
   range: 100
   rotation_period: 0.1
-  points_per_second: 56000
+  points_per_second: 10000
```
Comment (MishkaMN) [inline sensorlib/config/simulated_sensor_config.yaml:26]: Change back to 56000

File: sensorlib/src/objects/DetectedObject.py [hunk 8]
```diff
@@ -47,12 +51,18 @@ def build_detected_object(carla_actor, allowed_semantic_tags):
         if (bounding_box == None):
             return None
 
+        
+        projection_string = projection_string_config
+
+        #TODO: replace with correct size calculation
```
Comment (MishkaMN) [inline sensorlib/src/objects/DetectedObject.py:57]: Let's create github issue and assign write here

File: sensorlib/src/objects/DetectedObject.py [hunk 9]
```diff
@@ -64,5 +74,12 @@ def build_detected_object(carla_actor, allowed_semantic_tags):
             np.zeros((3, 3)),
             np.zeros((3, 3)),
 
-            1.0
+            1.0,
+            projection_string,
+            [size_x, size_y, size_z],
+            #TODO: replace with carla sensor timestamp
```
Comment (MishkaMN) [inline sensorlib/src/objects/DetectedObject.py:80]: Let's create github issue and assign write here

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Review by MishkaMN — CHANGES_REQUESTED** [review]

File: sensorlib/src/objects/DetectedObject.py [hunk 10]
```diff
@@ -15,28 +15,32 @@
...
+    velocityCovariance: np.ndarray
+    orientationCovariance: np.ndarray
+    angularVelocityCovariance: np.ndarray  
     confidence: float
+    projString: str
+    size: np.ndarray
+    timestamp: int
+    sensorId: str
+    carla_actor: carla.Actor  
+    bounding_box_in_world_coordinate_frame: List[np.ndarray]
```
Comment (MishkaMN) [inline sensorlib/src/objects/DetectedObject.py:38]: Let's search in the sensorlib directory for every occurences of these name changes and change them. On top of the id name change, I see changes required for object_type for example. Please double check

**Review by MishkaMN — CHANGES_REQUESTED** [review]

**Comment by MishkaMN** [comment]
One last leftover change needed:
`object_type` here needs to be `type` I think: https://github.com/usdot-fhwa-stol/carma-utils/blob/fix/detectObjects_fix/sensorlib/src/noise_models/GaussianNoiseModel.py#L65

**Review by MishkaMN — APPROVED** [review]
